src/components/collectible-icon/collectible-icon.tsx

```diff
@@ -2,14 +2,20 @@ import React, { FC, useState } from 'react';
 import { Image, View } from 'react-native';
 
 import { formatSize } from '../../styles/format-size';
-import { formatCollectibleUri } from '../../utils/image.utils';
+import { formatCollectibleUri, formatImgUri } from '../../utils/image.utils';
 import { isDefined } from '../../utils/is-defined';
 import { CollectibleIconProps } from './collectible-icon.props';
 import { useCollectibleIconStyles } from './collectible-icon.styles';
 
 export const CollectibleIcon: FC<CollectibleIconProps> = ({ collectible, size }) => {
   const styles = useCollectibleIconStyles();
   const [isLoaded, setIsLoaded] = useState(false);
+  const [isLoadingFailed, setIsLoadingFailed] = useState(false);
+
+  let uri;
+  if (isDefined(collectible) && isDefined(collectible.thumbnailUri)) {
+    uri = isLoadingFailed ? formatImgUri(collectible.thumbnailUri) : formatCollectibleUri(collectible);
+  }
```

Comment:
don't use `let` variables in React components  
if u have simple condition use `const` and `ternary operator`  
otherwise use React `useState`, `useMemo`, `useEffect`